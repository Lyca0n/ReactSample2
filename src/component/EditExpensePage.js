import React from 'react';

const EditExpensePage = (props) => {
    return (
        <div>
            This is the edit expense component with id of {props.match.params.id}
        </div>
    );
}

export default EditExpensePage;