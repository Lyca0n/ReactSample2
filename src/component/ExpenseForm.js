import React from 'react';
import moment from 'moment';
//import 'react-dates/initialize';
import { SingleDatePicker } from 'react-dates';



export default class ExpenseForm extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            description: props.expense ? props.expense.description:'',
            amount:  props.expense ? (props.expense.amount/100).toString() :'',
            note: props.expense ? props.expense.note:'',
            createdAt: props.expense ? new moment(props.expense.createdAt) : new moment(),
            focused: false,
            error:''
        };
    }

    onDescriptionChange = (e) => {
        const description = e.target.value;
        this.setState(() => ({ description }));
    };
    onNoteChange = (e) => {
        const note = e.target.value;
        this.setState(() => ({ note }));
    };
    onAmountChange = (e) => {
        const amount = e.target.value;
        if (! amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
            this.setState(() => ({ amount }));
        }
    };
    onDateChange = (createdAt) => {
        if(createdAt)
        this.setState(() => ({ createdAt }));
    };
    onFocusChange = ({focused}) => {
        this.setState(() => ({ focused }))
    };
    onSubmit=(e)=>{
        e.preventDefault();
        if(!this.state.amount || !this.state.description){
            this.setState(()=>({error : 'Provide descripton and amount'}));
        }else{
            this.setState(()=>({error : ''}));
            this.props.onSubmit({
                description:this.state.description,
                amount:parseFloat(this.state.amount,10)*100,
                createdAt: this.state.createdAt.valueOf(),
                note:this.state.note
            });
        }
    };
    render() {
        if (this.state.hasError) {
            // You can render any custom fallback UI
            return <h1>Something went wrong.</h1>;
        } else {
            return (
                <div>
                    <div >{this.state.error && <p>{this.state.error}</p>}</div>
                    <form onSubmit={this.onSubmit}>
                        <input
                            type="text"
                            placeholder="Description"
                            autoFocus
                            value={this.state.description}
                            onChange={this.onDescriptionChange}
                        />
                        <input
                            value={this.state.amount}
                            onChange={this.onAmountChange}
                            type="number"
                            placeholder="Amount"
                        />
                        <SingleDatePicker
                            date={this.state.createdAt}
                            onDateChange={this.onDateChange}
                            focused={this.state.focused}
                            onFocusChange={this.onFocusChange}
                            numberOfMonths={1}
                            isOutsideRange={(day)=>{false}}
                        />
                        <textarea
                            value={this.note}
                            onChange={this.onNoteChange}
                            placeholder="add optional note"
                        >
                        </textarea>
                        <button>Add Expense</button>
                    </form>

                </div>
            );
        }
    }
}