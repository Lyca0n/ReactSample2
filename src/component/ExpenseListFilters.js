import React from 'react';
import { connect } from 'react-redux';
//import 'react-dates/initialize';
import { DateRangePicker } from 'react-dates';
import { setTextFilter, sortByDate, sortByAmount,setStartDate, setEndDate } from '../actions/filters';

export class ExpenseListFilters extends React.Component {
    state = {
        focused: null
    };
    onDatesChange=({startDate, endDate})=>{
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    };
    onFocusChange=(focused)=>{
        this.setState(()=>({
            focused
        }));
    };  
    onTextChange= (e)=>{
        this.props.setTextFilter(e.target.value);
    };  
    onSortChange= (e) => {
        if (e.target.value === 'date') {
            this.props.sortByDate();
        } else if (e.target.value === 'amount') {
            this.props.sortByAmount();
        }
 
    };
    render() {
        return (<div>
            <input
                type="text"
                value={this.props.filters.text} onChange={this.onTextChange}
            />
            <select
                value={this.props.filters.sortBy} onChange={this.onSortChange}
            >
                <option value="date">Date</option>
                <option value="amount">Amount</option>
            </select>
            <DateRangePicker
                startDate={this.props.filters.startDate}
                startDateId='startDate'
                endDate={this.props.filters.endDate}
                endDateId='endDate'
                onDatesChange={this.onDatesChange}
                focusedInput={this.state.focused}
                onFocusChange={this.onFocusChange}
                showClearDates={true}
                numberOfMonths={1}
                isOutsideRange={(day)=>{false}}
            />
        </div>);
    }
}

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    };
};
const mapDispatchToProps=(dispatch)=>{
    return{
        setStartDate: (startDate)=>{
            dispatch(setStartDate(startDate))
        },
        setEndDate: (endDate)=>{
            dispatch(setEndDate(endDate))
        },
        sortByAmount:()=>{
            dispatch(sortByAmount)
        },
        sortByDate:()=>{
            dispatch(sortByDate)
        },
        setTextFilter:(text)=>{
            dispatch(setTextFilter(text));
        }
    };
};
export default connect(mapStateToProps,mapDispatchToProps)(ExpenseListFilters);