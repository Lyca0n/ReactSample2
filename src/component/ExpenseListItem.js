import React from 'react';
import {connect} from 'react-redux';
import { removeExpense } from '../actions/expenses';

const ExpenseListItem =({dispatch, id, description,amount,createdAt})=>(
    <div>
        {description && <h1>description: {description}</h1>}
        {amount && <p>Amount: {amount}</p>}
        {createdAt && <p>Created at: {createdAt}</p>}  
        <button onClick={(e)=>{
            dispatch(removeExpense({id}));
        }}>Remove Expense</button>      
    </div>
);

export default connect()(ExpenseListItem);