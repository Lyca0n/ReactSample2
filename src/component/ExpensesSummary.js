import  React from 'react';
import numeral from 'numeral';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({expenseCount, expenseTotal})=>{
    const expenseWord = expenseCount ===1? 'expense' :'expenses';
    const formatTotal = numeral(expenseTotal/100).format('$0,0.00');
    return (
    <div > 
        <h2>{'Viewing '+expenseCount+' '+expenseWord+' totalling '+ formatTotal }</h2>
    </div>
    );
};

const mapStateToProps=(state)=>{
    const vis =selectExpenses(state.expenses, state.filters);
    return{
        count: vis.length,
        total: selectExpensesTotal(vis)
    };
};
export default connect(mapStateToProps)(ExpensesSummary);