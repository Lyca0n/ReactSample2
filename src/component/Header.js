import React from 'react';
import {NavLink} from 'react-router-dom';
import { connect } from 'react-redux';
import {startLogout} from '../actions/auth';

export const Header = ({startLogout}) => (
    <header>
        <h1>Expensify</h1>
        <ul>
            <li>
                <NavLink activeClassName="is-active" to="/dashboard">Dashboard</NavLink>
            </li>
            <li>
                <NavLink activeClassName="is-active" to="/create">Create Expenses</NavLink>
            </li>
            <li>
                <button onClick={startLogout}>Logout</button>
            </li>
        </ul>
    </header>
);

const mapDispatchToProps=(dispatch)=>({
    startLogout: ()=> dispatch(startLogout())
});

export default connect(undefined,mapDispatchToProps)(Header);