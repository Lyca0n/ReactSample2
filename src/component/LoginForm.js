import React from 'react';

export default class LoginForm extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            username: '',
            password: '',
            error:''
        };
    }

    onUsernameChange = (e) => {
        const username = e.target.value;
        this.setState(() => ({ username }));
    };
    onPasswordChange = (e) => {
        const password = e.target.value;
        this.setState(() => ({ password }));
    };

    onSubmit=(e)=>{
        e.preventDefault();
        if(!this.state.username || !this.state.password){
            this.setState(()=>({error : 'Provide descripton and amount'}));
        }else{
            this.setState(()=>({error : ''}));
            this.props.startLogin({});
        }
    };
    render() {
        if (this.state.hasError) {
            // You can render any custom fallback UI
            return <h1>Something went wrong.</h1>;
        } else {
            return (
                <div>
                    <div >{this.state.error && <p>{this.state.error}</p>}</div>
                    <form onSubmit={this.onSubmit}>
                        <input
                            type="text"
                            placeholder="Username"
                            autoFocus
                            value={this.state.username}
                            onChange={this.onUsernameChange}
                        />
                        <input
                            value={this.state.password}
                            onChange={this.onPasswordChange}
                            type="password"
                            placeholder="Password"
                        />
                        <button>Login</button>
                    </form>

                </div>
            );
        }
    }
}