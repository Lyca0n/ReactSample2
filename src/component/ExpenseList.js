import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';

const ExpenseList= (props)=>(
    <div>
        <h1>ExpenseList</h1>
        {props.expenses.map((expense)=>{
            return <ExpenseListItem key={expense.id} {...expense}/>
        })}
    </div>
);

//map function
const mapStateToProps =(state)=>{
    return {
        expenses: state.expenses
    };
};

//"connect()"" returns a function to recieve the component name
export default connect(mapStateToProps)(ExpenseList);
