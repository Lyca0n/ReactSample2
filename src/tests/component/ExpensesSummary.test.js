import React  from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import {ExpensesSummary} from '../../component/ExpensesSummary';


test('Should render ExpensesSummary component correctly single expense',()=>{
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expenseTotal={9434}/>);
    expect(wrapper).toMatchSnapshot();
});
test('Should render ExpensesSummary component correctly multiple expense',()=>{
    const wrapper = shallow(<ExpensesSummary expenseCount={34} expenseTotal={9434}/>);
    expect(wrapper).toMatchSnapshot();
});
