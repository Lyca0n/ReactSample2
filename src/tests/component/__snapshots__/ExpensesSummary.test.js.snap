// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpensesSummary component correctly multiple expense 1`] = `
<div>
  <h2>
    Viewing 34 expenses totalling $94.34
  </h2>
</div>
`;

exports[`Should render ExpensesSummary component correctly single expense 1`] = `
<div>
  <h2>
    Viewing 1 expense totalling $94.34
  </h2>
</div>
`;
